src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -151,6 +160,16 @@ class BlockchainImpl(
 
   override def getTotalDifficultyByHash(blockhash: ByteString): Option[BigInt] = totalDifficultyStorage.get(blockhash)
 
+  override def getAccount(address: Address, blockNumber: BigInt): Option[Account] =
+    getBlockHeaderByNumber(blockNumber).flatMap { bh =>
+      val mpt = new MerklePatriciaTrie[Address, Account](
```

Comment:
That `hashFn` is used to configure the function used to hash the encoded nodes for storing them in the `nodeStorage` (so it's a used as a key-value storage as `hashFn(encode(node))->encode(node)`, which is different from the key-value map `hashFn2(address)->account` that is the MPT created). Although in Ethereum the hash function used for the storage is always `kec256`, it is configurable on the MPT.